### PR TITLE
feat(opsim): create depositTx subscriber

### PR DESCRIPTION
**Pull request by jakim929**
- `SubscribeDepositTx` creates a subscription to depositTransactions given a chain api and the deposit contract address. under the hood it transforms the event log subscription
- adds `chainapi` package, which is the interface that chains should expose to opsim 
- adds some testutils `MockChain` which can be overridden with methods for testing

this is not used anywhere right now, but will be used in a separate PR once we figure out how to get bridge contracts deployed on the L1

it will be used like below
```

func (opSim *OpSimulator) StartDepositProcessor(ctx context.Context) error {
	depositTxCh := make(chan *types.DepositTx)

	sub, err := SubscribeDepositTx(ctx, opSim.l1Chain, depositContractAddr, depositTxCh)

	if err != nil {
		return fmt.Errorf("failed to start deposit processor: %w", err)
	}

	go func() {
		for {
			select {
			case dep := <-depositTxCh:
				if err := opSim.l2Chain.EthSendTransaction(ctx, types.NewTx(dep)); err != nil {
					opSim.log.Error("failed to submit deposit tx: %w", err)
				}
			case <-ctx.Done():
				sub.Unsubscribe()
				close(depositTxCh)
				return
			}
		}
	}()

	return nil
}

```

### CODE REVIEW

**Comment by jakim929** [comment]
* **#45** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/supersim/45?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jakim929 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: anvil/anvil.go
Comment (jakim929) [inline anvil/anvil.go:175]: wsEndpoint supports subscriptions

File: opsimulator/deposits.go
Comment (tremarkley) [inline opsimulator/deposits.go:27]: did you mean to leave this `Println` in?

**Review by tremarkley — APPROVED** [review]
this all looks great to me, but I would probably wait on a review from @nitaliano or @hamdiallam before merging

File: anvil/anvil.go
Comment (nitaliano) [inline anvil/anvil.go:182]: super nit but this can be one line

**Review by nitaliano — APPROVED** [review]
nice!

**Pull request merged**
